src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/db/repositories/ProbationDeliveryUnitRepository.kt

```diff
@@ -5,5 +5,5 @@ import org.springframework.data.repository.kotlin.CoroutineCrudRepository
 import uk.gov.justice.digital.hmpps.hmppsprobationestateapi.db.entities.ProbationDeliveryUnit
 
 interface ProbationDeliveryUnitRepository : CoroutineCrudRepository<ProbationDeliveryUnit, String> {
-  suspend fun findByRegionCode(regionCode: String): Flow<ProbationDeliveryUnit>
+  suspend fun findByRegionCodeAndSoftDeletedFalse(regionCode: String): Flow<ProbationDeliveryUnit>
```

Comment:
It might be worth considering a re-design in future - these cross-cutting concerns may be better enforced by types. So the DB entities are never sent back (good design BTW!), which means we can filter out inactive/soft-deleted items when transforming.

It might not be much better, but it would put this don't-use-inactive-items logic in the code rather than having to be part of every DB call/repo method.

Reply:
JPA has better things out the box but unfortunately spring data r2dbc doesn't at the moment